### PR TITLE
feat(@angular/cli): making app root configurable

### PR DESCRIPTION
**Pull request by sumitarora**
Fixes: https://github.com/angular/angular-cli/issues/5927

### CODE REVIEW

File: packages/@angular/cli/blueprints/component/index.ts [hunk 1]
```diff
@@ -112,7 +112,8 @@ export default Blueprint.extend({
         resolveModulePath(options.module, this.project, this.project.root, appConfig);
     } else {
       try {
-        this.pathToModule = findParentModule(this.project.root, appConfig.root, this.generatePath);
+        this.pathToModule = findParentModule
```
Comment (hansl) [inline packages/@angular/cli/blueprints/component/index.ts]: Parentheses on same line as function. Same everywhere.

**Comment by sumitarora** [comment]
@Brocco Added Test 😄

**Comment by clydin** [comment]
Looking at the blueprint, what is currently not configurable is the main ng module name (I.e, the bootstrapped module) and component name.
The "root" setting really is the app root as a large amount of app specific files are located there (index.html, main.ts, etc.).

**Comment by Brocco** [comment]
Can you please rebase & repush this PR? There's a lot of noise in the diffs to review.

**Comment by sumitarora** [comment]
@hansl @Brocco Updated and rebased 😄

**Comment by hansl** [comment]
Please rebase.

**Comment by hansl** [comment]
Closing every Blueprints related issue and PR.

**Pull request closed without merging**